jaeger-core/src/test/java/io/jaegertracing/internal/propagation/TextMapCodecTest.java

```diff
@@ -44,4 +50,55 @@ public void testWithoutBuilder() {
     assertTrue(str.contains("urlEncoding=false"));
   }
 
+  @Test
+  public void testSpanToString() {
+    JaegerSpanContext expectedContext = new JaegerSpanContext(1, 2, 3, (byte) 1);
+    JaegerSpanContext actualContext = TextMapCodec.contextFromString(expectedContext.contextAsString());
+
+    assertEquals(expectedContext.getTraceId(), actualContext.getTraceId());
+    assertEquals(expectedContext.getSpanId(), actualContext.getSpanId());
+    assertEquals(expectedContext.getParentId(), actualContext.getParentId());
+    assertEquals(expectedContext.getFlags(), actualContext.getFlags());
+  }
+
+  @Test(expected = MalformedTracerStateStringException.class)
+  public void testContextFromStringMalformedException() throws Exception {
+    TextMapCodec.contextFromString("ff:ff:ff");
+  }
+
+  @Test(expected = EmptyTracerStateStringException.class)
+  public void testContextFromStringEmptyException() throws Exception {
+    TextMapCodec.contextFromString("");
+  }
+
+  @Test
+  public void testContextFromString() throws Exception {
+    JaegerSpanContext context = TextMapCodec.contextFromString("ff:dd:cc:4");
+    assertEquals(context.getTraceId(), 255);
+    assertEquals(context.getSpanId(), 221);
+    assertEquals(context.getParentId(), 204);
+    assertEquals(context.getFlags(), 4);
+  }
+
+  @Test
+  public void testToStringFormatsPositiveFields() {
+    long traceId = -10L;
+    long spanId = -10L;
+    long parentId = -10L;
+    byte flags = (byte) 129;
+
+    // I use MIN_VALUE because the most significant bit, and thats when
+    // we want to make sure the hex number is positive.
```

Comment:
this comment looks like some old left-over. Doesn't seem to make any sense now.

Reply:
Good catch.